packages/m2m-gateway/src/services/eserviceService.ts

```diff
@@ -57,6 +64,25 @@ export function eserviceServiceBuilder(
     };
   };
 
+  const pollEServiceById = (
+    eserviceId: EServiceId,
+    metadata: { version: number } | undefined,
+    headers: M2MGatewayAppContext["headers"]
+  ): Promise<WithMaybeMetadata<catalogApi.EService>> =>
+    pollResourceWithMetadata(() => retrieveEServiceById(headers, eserviceId))({
+      condition: isPolledVersionAtLeastMetadataTargetVersion(metadata),
+    });
+
+  const pollEService = (
```

Comment:
very minor: a bit confusing that both this and `pollEServiceById` exist. 
We could call required the function like this
```
pollEService(unsafeBrandId(response.data.id), headers)
```